### PR TITLE
Roll engine to version b148e628ec86b3a9a0382e0bcfae73f0390a8232

### PR DESCRIPTION
**Pull request by mkustermann**
This includes:

   Roll Dart to version be6309690fd60284a87f3258a740c7c30efb1092 (#5987)

### CODE REVIEW

**Review by mraleph — APPROVED** [review]

**Comment by aam** [comment]
There seems to be issues with mockito package:
```
╰─➤  bin/flutter doctor                                                                                                               130 ↵
Building flutter tool...
The current Dart SDK version is 2.1.0-dev.0.0.flutter-be6309690f.

Because flutter_tools depends on mockito >=0.8.1 <3.0.0-beta+3 which requires SDK version >=1.0.0 <2.0.0-∞, version solving failed.
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds...
```

**Comment by mkustermann** [comment]
Turns out Siva's roll in the engine wasn't further rolled into flutter/flutter yet.

Doing this will require upgrading all 3rd party packages in pubspec files to be compatible with Dart 2, since this roll includes a new version number, namely `2.1.0-dev.0.0`.

I've tried to find all the necessary package upgrades and filed issues for:

* https://github.com/google/process.dart/issues/32
* https://github.com/google/platform.dart/issues/17
* https://github.com/google/file.dart/issues/105

So far.

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 1]
```diff
@@ -27,9 +27,9 @@ import '../runner/flutter_command.dart';
 /// ```
 const Map<String, String> _kManuallyPinnedDependencies = <String, String>{
   // Add pinned packages here.
-  'mockito': '3.0.0-beta', // TODO(aam): https://github.com/dart-lang/mockito/issues/110
-  'matcher': '0.12.2+1', // TODO(ianh): https://github.com/flutter/flutter/issues/18608, https://github.com/dart-lang/matcher/pull/88
-  'flutter_gallery_assets': '0.1.4', // See //examples/flutter_gallery/pubspec.yaml
+  'mockito': '3.0.0',
+  'matcher': '0.12.3+1', // TODO(ianh): https://github.com/flutter/flutter/issues/18608
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: Please don't do this. It's a breaking change for our users.

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 2]
```diff
@@ -1075,6 +1075,7 @@ String _generateFakePubspec(Iterable<PubspecDependency> dependencies) {
     if (!dependency.pointsToSdk)
       dependency.describeForFakePubspec(result, overrides);
   result.write(overrides.toString());
+  //print('XXXX: ${result.toString()}');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: nit: commented out code

Comment (mkustermann) [reply packages/flutter_tools/lib/src/commands/update_packages.dart]: Done.

File: packages/flutter_tools/lib/src/commands/update_packages.dart (hunk 1 above)
Comment (mkustermann) [reply packages/flutter_tools/lib/src/commands/update_packages.dart]: Guess this issue has been resolved already by https://github.com/flutter/flutter/pull/20602

**Comment by mkustermann** [comment]
The SEGV crash in the `flutter_tester` binary crashed very deeply inside the Dart VM when doing a constant evaluation inside `package:test` code.

After debugging this for a while it became clear that this is due to a missing platform file, which caused the VM to fall back to code paths which it should never fall back to in a Dart 2.0 world.

The root cause was flutter/engine@a84b210b3d26513522d65060456517aee3335b20 which renamed the platform file the `flutter_tester` binary expects, but a corresponding change in `flutter` wasn't done yet.

**Comment by mkustermann** [comment]
Since the engine->flutter roll also brings in new skia changes, I've made https://github.com/flutter/goldens/pull/11 to update the goldens.

**Comment by mraleph** [comment]
still LGTM

**Comment by mkustermann** [comment]
The existing redness on the flutter dashboard needs to be investigated, e.g. I can see e.g. `stderr: > Java heap space` in [this log](https://flutter-dashboard.appspot.com/api/get-log?ownerKey=ahNzfmZsdXR0ZXItZGFzaGJvYXJkclgLEglDaGVja2xpc3QiOGZsdXR0ZXIvZmx1dHRlci9mYWZmZDNhZWYzMDM1Y2JlMmYyMDVlZWE0YjNmNzJhNTI5NWRkNWU4DAsSBFRhc2sYgICAgICAzAgM), but I think it makes sense to land this now, since there are already more engine commits waiting to be rolled in.

**Pull request merged**